cse-lib/data/src/main/java/com/farao_community/farao/cse/data/ttc_res/TtcResult.java

```diff
@@ -502,12 +507,14 @@ private static void fillActivatedPreventiveRemedialActions(CracResultsHelper cra
             action.setName(name);
             actionList.add(action);
         });
-        pstPreventiveRas.forEach(parade -> {
+        pstPreventiveRas.forEach(jsonRaId -> {
             Action action = new Action();
             Name name = new Name();
-            name.setV(parade);
+            CsePstCreationContext csePstCreationContext = getPstCreationContext(cracResultsHelper.getCseCracCreationContext(), jsonRaId);
+            name.setV(csePstCreationContext.getNativeId());
             PSTtap pstTap = new PSTtap();
-            pstTap.setV(BigInteger.valueOf(cracResultsHelper.getTapOfPstRangeActionInPreventive(parade)));
+            int tap = csePstCreationContext.isInverted() ? -cracResultsHelper.getTapOfPstRangeActionInPreventive(jsonRaId) : cracResultsHelper.getTapOfPstRangeActionInPreventive(jsonRaId);
+            pstTap.setV(BigInteger.valueOf(tap));
             action.setPSTtap(pstTap);
```

Comment:
I think you can invert the logic for remedial acion as well, do `cracResultsHelper.getCseCracCreationContext().getRemedialActionCreationContexts().stream().filter(ElementaryCreationContext::isImported)` and then apply the proper treatment for each. It will avoid to do a loop everytime across creation context to find the good one i.e. `getPstCreationContext` method